examples/chef/common/chef-fan-control-manager.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/clusters/fan-control-server/fan-control-server.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/error-mapping.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::FanControl;
+using namespace chip::app::Clusters::FanControl::Attributes;
+using Protocols::InteractionModel::Status;
+
+namespace {
+class ChefFanControlManager : public AttributeAccessInterface, public Delegate
+{
+public:
+    // Register for the FanControl cluster on all endpoints.
```

Comment:
the comment says `all endpoints` yet the constructor takes a `endpointId` as an argument. What is it?

Reply:
It means that the we should it will be registered to every endpoint when the ep's fan control cluster init is called.

```
void emberAfFanControlClusterInitCallback(EndpointId endpoint)
{
    VerifyOrDie(mFanControlManager == nullptr);
    mFanControlManager = new ChefFanControlManager(endpoint);
    registerAttributeAccessOverride(mFanControlManager);
    FanControl::SetDefaultDelegate(endpoint, mFanControlManager);
}
```

Reply:
You need to re-phrase it somehow. This comment did not help me understand what it means or how code is supposed to work. Maybe this is a "register this fan control for this particular endpoint" (which is actually obvious from the signature) or a note saying "It is generally expected that every endpoint will have its own copy of a fan control manager".

Overall I think we should just remove this comment.

Reply:
Comment removed.